### PR TITLE
added unit test for p5.gain in es6

### PR DESCRIPTION
**Pull request by endurance21**
added unit tests for all the methods in p5.Gain 
here are the test  results and AUDIO GRAPH ( inspected with WEB AUDIO INSPECTOR) of this unit test 
Selection_086

### CODE REVIEW

**Comment by endurance21** [comment]
@therewasaguy  
please review this PR ! 
Also, i have been using this amazing `WEB AUDIO INSPECTOR`  tool to see the AUDIO GRAPHS 
, I find it very useful while working with web audio API, i am curious if we can use this as integrated part of our codebase, it will really help the developers to understand and write codes ,specially when they can see what they are creating!

**Comment by endurance21** [comment]
@therewasaguy we can add it too?

File: test/tests/p5.Gain.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            let gain = new p5.Gain();
+            gain.dispose();
+        });
+        it('can set input', ()=>{
+            let inputNode = new p5.Gain();
+            let mainGainNode  = new p5.Gain();
+
+            mainGainNode.setInput(inputNode);
+        });
+        it('can send output',()=>{
```
Comment (therewasaguy) [inline test/tests/p5.Gain.js:16]: this test is doing the same thing as 'can set input' maybe they can just be one test?

File: test/tests/p5.Gain.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            mainGainNode.connect(outputNode);
+        });
+        it('can disconnect  all output' ,()=>{
+            let inputNode = new p5.Gain();
+            let outputNode = new p5.Gain();
+            let mainGainNode  = new p5.Gain();
+
+            mainGainNode.setInput(inputNode);
+            mainGainNode.connect(outputNode);
+            mainGainNode.disconnect();
```
Comment (therewasaguy) [inline test/tests/p5.Gain.js:29]: this is testing that we are able to call the `disconnect` method without that throwing any runtime errors, but how do we know if we successfully disconnected? (I'm not sure!)

Also, will this disconnect the inputNode or only the connection between the mainGaneNode and the outputNode? I think it's only that one. Again, how would we know from this test?

File: test/tests/p5.Gain.js (hunk 1 above)
Comment (endurance21) [reply test/tests/p5.Gain.js:16]: though they appear to do the same task but actually we are able to validate twoo methods here 
setInput() and Connect() method

Comment (therewasaguy) [reply test/tests/p5.Gain.js:16]: ahh, got it! Thank you for clarifying and sorry I missed this

**Comment by Abhijay007** [comment]
Hi @endurance21,  can you please look into the merge conflicts, if you are busy I can resolve those conflicts and make a new PR around the same :)

**Review by Abhijay007 — COMMENTED** [review]
Seems like the Test for p5.Gain.js is already updated and present in the repo, refer: https://github.com/processing/p5.js-sound/blob/main/test/tests/p5.Gain.js (last updated by @satyasaibhushan,  

@davepagurek I think we can close this PR now.

**Comment by davepagurek** [comment]
We may want to copy over just the `can set the output level of gain Node` test, since the current test file doesn't test changing the gain at all. But I think that's the only one that isn't covered now!

**Comment by Abhijay007** [comment]
> We may want to copy over just the `can set the output level of gain Node` test, since the current test file doesn't test changing the gain at all. But I think that's the only one that isn't covered now!

okay got it, so can I open a PR to add the `can set the output level of gain Node`  in the existing gain test?

**Comment by davepagurek** [comment]
yeah sounds good!

**Comment by Abhijay007** [comment]
Okay will do !!

**Comment by davepagurek** [comment]
Thanks @Abhijay007! I'll close this PR and carry over reviews on your new one.

**Pull request closed without merging**